// @testomatio/reporter/lib/reporter.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.linkJira = exports.linkTest = exports.label = exports.step = exports.meta = exports.logger = exports.log = exports.artifact = void 0;
// import TestomatClient from './client.js';
// import * as TRConstants from './constants.js';
const index_js_1 = require("./services/index.js");
const reporter_functions_js_1 = __importDefault(require("./reporter-functions.js"));
exports.artifact = reporter_functions_js_1.default.artifact;
exports.log = reporter_functions_js_1.default.log;
exports.logger = index_js_1.services.logger;
exports.meta = reporter_functions_js_1.default.keyValue;
exports.step = reporter_functions_js_1.default.step;
exports.label = reporter_functions_js_1.default.label;
exports.linkTest = reporter_functions_js_1.default.linkTest;
exports.linkJira = reporter_functions_js_1.default.linkJira;
/**
 * @typedef {typeof import('./reporter-functions.js').default.artifact} ArtifactFunction
 * @typedef {typeof import('./reporter-functions.js').default.log} LogFunction
 * @typedef {typeof import('./services/index.js').services.logger} LoggerService
 * @typedef {typeof import('./reporter-functions.js').default.keyValue} MetaFunction
 * @typedef {typeof import('./reporter-functions.js').default.step} StepFunction
 * @typedef {typeof import('./reporter-functions.js').default.label} LabelFunction
 */
module.exports = {
    /**
     * @deprecated Use `log` or `testomat.log`
     */
    testomatioLogger: index_js_1.services.logger,
    artifact: reporter_functions_js_1.default.artifact,
    log: reporter_functions_js_1.default.log,
    logger: index_js_1.services.logger,
    meta: reporter_functions_js_1.default.keyValue,
    step: reporter_functions_js_1.default.step,
    label: reporter_functions_js_1.default.label,
    linkTest: reporter_functions_js_1.default.linkTest,
    linkJira: reporter_functions_js_1.default.linkJira,
    // TestomatClient,
    // TRConstants,
};
